expression/builtin_control.go

```diff
@@ -114,9 +122,9 @@ func InferType4ControlFuncs(lexp, rexp Expression) *types.FieldType {
 			if lhs.Decimal != types.UnspecifiedLength {
 				rhsFlen -= rhs.Decimal
 			}
-			resultFieldType.Flen = mathutil.Max(lhsFlen, rhsFlen) + resultFieldType.Decimal + 1
+			resultFieldType.Flen = maxlen(lhsFlen, rhsFlen) + resultFieldType.Decimal + 1
```

Comment:
Will the `resultFieldType.Flen` exceed `MaxDecimalWidth`?

Reply:
I am not sure, but I can wrap it in `mathutil.Min()`, since `maxlen` follows an arbitrary size observed in MySQL:
```
                        }
-                       resultFieldType.Flen = maxlen(lhsFlen, rhsFlen) + resultFieldType.Decimal + 1
+                       flen := maxlen(lhsFlen, rhsFlen) + resultFieldType.Decimal + 1   // account for -1 len fields
+                       resultFieldType.Flen = mathutil.Min(flen, mysql.MaxDecimalWidth) // make sure it doesn't overflow
```
What do you think?

Reply:
Yeah. Wrapping it in `mathutil.Min()` is feasible to me.